src/smart.js

```diff
@@ -1,30 +1,21 @@
 /* global window */
-const {
+import {
     isBrowser,
-    debug: _debug,
+    debug as _debug,
     request,
     getPath,
     randomString,
     btoa,
     getAndCache
-} = require("./lib");
+} from "./lib";
 
-const debug = _debug.extend("oauth2");
-const { SMART_KEY } = require("./settings");
+import Client from "./Client";
 
+import { SMART_KEY } from "./settings";
 
-/**
- * Creates and returns a Client instance.
- * Note that this is done within a function to postpone the "./Client" import
- * and avoid cyclic dependency.
- * @param {fhirclient.JsonObject} env The adapter
- * @param {string | fhirclient.ClientState} state The client state or baseUrl
- * @returns {fhirclient.Client}
- */
-function createClient(env, state) {
-    const Client  = require("./Client");
-    return new Client(env, state);
-}
```

Comment:
I vaguely remember that the reason for the existence of the (otherwise useless) `createClient` function was to avoid some kind of circular dependency, e.g.: `Client.js` and `smart.js` require each other. Perhaps this is not an issue for ESM but I have to ask - have you had any problems with that?

Reply:
ESM can resolve the circular import by the time the other functions in this file need it. It was necessary as CJS code prior to the conversion.